ipynb/agp_distance_decay.ipynb

```diff
@@ -0,0 +1,279 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now that our data are successfully loaded, let's compute our distance matrices."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "## compute Canberra distances\n",
+    "comm.dist <- vegdist(t(dat.rel.trans), method = 'canberra')\n",
+    "comm.dist <- as.matrix(comm.dist)\n",
+    "\n",
+    "## compute great circle distance for sites\n",
+    "gc.dist <- rdist.earth(coords, miles = FALSE, R = 6371) ## manually added a more accurate estimate of Earth's radius\n",
+    "colnames(gc.dist) <- colnames(comm.dist)\n",
+    "rownames(gc.dist) <- rownames(comm.dist)\n",
+    "\n",
+    "## this is a pretty large dataset, so let's save some memory and\n",
+    "## just plot the upper triangles of both matrices to avoid duplicate points\n",
+    "## in the scatterplot\n",
+    "tri.comm <- comm.dist\n",
+    "diag(tri.comm) <- NA\n",
+    "tri.comm[lower.tri(tri.comm)] <- NA\n",
+    "\n",
+    "tri.gc <- gc.dist\n",
+    "diag(tri.gc) <- NA\n",
+    "tri.gc[lower.tri(tri.gc)] <- NA\n",
+    "\n",
+    "## bind data for plotting\n",
+    "ggplot.dat <- na.omit(data.frame('comm' = c(tri.comm), 'gc' = c(tri.gc)))\n",
+    "\n",
+    "## distance-decay plot for all data\n",
+    "ggplot(ggplot.dat, aes(x = gc, y = comm)) + \n",
+    "  geom_point(colour = 'dodgerblue4', alpha = 0.7, size = 0.35) +\n",
+    "  stat_smooth(colour = 'black', method = \"lm\", \n",
+    "              formula = y ~ poly(x, 1)) + ## fits a linear model\n",
+    "  guides(colour = FALSE, fill = FALSE) +\n",
+    "  xlab('Spatial Distance (km)') +\n",
+    "  ylab('Community Distance') +\n",
+    "  theme_bw() + \n",
+    "  theme(panel.grid.minor = element_blank(), panel.grid.major = element_blank(), \n",
+    "        panel.border = element_blank(), panel.grid.minor = element_blank(), \n",
+    "        panel.background = element_blank()) +\n",
+    "  theme(axis.line = element_line(color = 'black'))\n",
+    "\n",
+    "## and we perform a permutational test to assess significance\n",
+    "mant.mod <- mantel(comm.dist, gc.dist, method = 'spear', permutations = 999)\n",
+    "mant.mod"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The next section performs individual analyses at the state-level, and performs Benjamini-Hochberg correction for multiple comparisons and spits them out into a summary table, so that you can evaluate which states (if any) display significant distance-decay relationships and plot those particular states, if you want.\n",
+    "\n",
+    "In the first version of the data I recieved, I had to manually clean up multiple discrepancies in state namings (e.g., state labels for Colorado were inputted as both 'CO' and 'Colorado'. Not sure if this has been corrected in the new data. If not, let me know and we can figure out a quick solution."
```

Comment:
That sounds good. Is there anything I can do on my end to help?

AF

On Mon, Jan 25, 2016 at 9:02 AM, J W Debelius <notifications@github.com>
wrote:

> In ipynb/agp_distance_decay.ipynb
> <https://github.com/biocore/American-Gut/pull/191#discussion_r50722181>:
>
> > +    "        panel.border = element_blank(), panel.grid.minor = element_blank(), \n",
> > +    "        panel.background = element_blank()) +\n",
> > +    "  theme(axis.line = element_line(color = 'black'))\n",
> > +    "\n",
> > +    "## and we perform a permutational test to assess significance\n",
> > +    "mant.mod <- mantel(comm.dist, gc.dist, method = 'spear', permutations = 999)\n",
> > +    "mant.mod"
> > +   ]
> > +  },
> > +  {
> > +   "cell_type": "markdown",
> > +   "metadata": {},
> > +   "source": [
> > +    "The next section performs individual analyses at the state-level, and performs Benjamini-Hochberg correction for multiple comparisons and spits them out into a summary table, so that you can evaluate which states (if any) display significant distance-decay relationships and plot those particular states, if you want.\n",
> > +    "\n",
> > +    "In the first version of the data I recieved, I had to manually clean up multiple discrepancies in state namings (e.g., state labels for Colorado were inputted as both 'CO' and 'Colorado'. Not sure if this has been corrected in the new data. If not, let me know and we can figure out a quick solution."
>
> The state names are corrected in the new version, so we shouldn't need to
> update. But, we're currently storing state information for states outside
> of the US and that can give weird results.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/American-Gut/pull/191/files#r50722181>.
>